### PR TITLE
Fix broken link in rustc_plugin doc

### LINKED ISSUE #57489: Broken link to unstable book in rustc_plugin API docs
The [documentation of rustc_plugin](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_plugin/index.html) links to the plugin feature in the unstable book, but that link is broken.

It points to:
https://doc.rust-lang.org/nightly/nightly-rustc/unstable-book/language-features/plugin.html

While the correct link would be:
https://doc.rust-lang.org/nightly/unstable-book/language-features/plugin.html

In the source, the link is relative:
`../unstable-book/language-features/plugin.html`

I'd be interested in fixing this, but I don't know whether a relative or absolute link is better in this situation.

### PR DESCRIPTION
**Pull request by agnxy**
fix #57489
r? @steveklabnik

### CODE REVIEW

File: src/librustc_plugin/lib.rs [hunk 1]
```diff
@@ -47,8 +47,9 @@
 //! #![plugin(myplugin)]
 //! ```
 //!
-//! See the [`plugin` feature](../unstable-book/language-features/plugin.html) of
-//! the Unstable Book for more examples.
+//! See the [`plugin`
+//! feature](https://doc.rust-lang.org/nightly/unstable-book/language-features/plugin.html)
```
Comment (lzutao) [inline src/librustc_plugin/lib.rs]: Replace the link with relative link: `../../unstable-book/language-features/plugin.html`

**Comment by Centril** [comment]
I don't know what the policy around linking is; r? @ehuss

File: src/librustc_plugin/lib.rs (hunk 1 above)
Comment (agnxy) [reply src/librustc_plugin/lib.rs]: Thanks @lzutao . I also prefer relative links. But I'm not sure whether `../../unstable-book/language-features/plugin.html` will affect offline documents. (See https://github.com/rust-lang/rust/issues/57489#issuecomment-482933292)

Comment (lzutao) [reply src/librustc_plugin/lib.rs]: What is local doc by the way? Is it distributed by Rust or you have to build
it locally?

Comment (agnxy) [reply src/librustc_plugin/lib.rs]: I guess the doc distributed by Rust doesn't include [rustc internal doc](https://doc.rust-lang.org/nightly/nightly-rustc/rustc/), which is to be fixed in this PR. So I have to build it for a local copy.
The unstable book is distributed with rustup.

**Comment by ehuss** [comment]
Considering that these aren't distributed with rustup (AFAIK), and that these are only published on nightly, I'm inclined to just use the full https link.  In the rare case someone is working offline, I think they'll be able to figure out what to do.

@bors r+

**Pull request merged**